product_contract/models/sale_order_line.py

```diff
@@ -250,20 +247,28 @@ def _compute_name(self):
                     self._fields["recurring_invoicing_type"].get_description(self.env)
                 )
                 field_selection = dict(field_info.get("selection"))
-                invoicing_type_label = field_selection.get(
-                    line.recurring_invoicing_type
-                )
-                line.name = _(
-                    """{product}:
-    - Recurrency: {recurring_rule}
-    - Invoicing Type: {invoicing_type}
-    - Date: {date_text}
-                """
-                ).format(
-                    product=line.product_id.display_name,
-                    recurring_rule=recurring_rule_label,
-                    invoicing_type=invoicing_type_label,
-                    date_text=date_text,
-                )
+                invoicing_type_label = field_selection.get(line.recurring_invoicing_type)
 
+                # Hook call here
+                line.name = line._get_contract_name_format(
+                    date_text, recurring_rule_label, invoicing_type_label
+                )
         return res
+
+
+    def _get_contract_name_format(self, date_text, recurring_rule_label, invoicing_type_label):
+        """Hook method to format the final contract name string."""
+        self.ensure_one()
+        return _(
+            """{product}:
+        - Recurrency: {recurring_rule}
+        - Invoicing Type: {invoicing_type}
+        - Date: {date_text}
+            """
+        ).format(
+            product=self.product_id.display_name,
+            recurring_rule=recurring_rule_label,
+            invoicing_type=invoicing_type_label,
+            date_text=date_text,
+        )
```

Comment:
The method name would be better renamed `_get_contract_name` or `_get_contract_description`

As it is not getting only the format but also assigns the values.

It might be even better to have a method that gives the only


```
_(
            """{product}:
        - Recurrency: {recurring_rule}
        - Invoicing Type: {invoicing_type}
        - Date: {date_text}
            """
        )
```
        
Something like:

```suggestion
    def _get_contract_name(self, date_text, recurring_rule_label, invoicing_type_label):
        self.ensure_one()
        return self._get_contract_name_format().format(
            product=self.product_id.display_name,
            recurring_rule=recurring_rule_label,
            invoicing_type=invoicing_type_label,
            date_text=date_text,
        )

    def _get_contract_name_format(self, date_text, recurring_rule_label, invoicing_type_label):
        self.ensure_one()
        return _(
            """{product}:
        - Recurrency: {recurring_rule}
        - Invoicing Type: {invoicing_type}
        - Date: {date_text}
            """
        )
```

Reply:
@yvaucher can you check now